### PR TITLE
feat!: add support for settings.active_directory_config for SQL module

### PR DESCRIPTION
**Pull request by vponnam**
This change beings support for connecting SQL servers to Managed AD domain.

New feature available in [v4.22.0 provider](https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/sql_database_instance#domain)

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: modules/mssql/variables.tf [hunk 1]
```diff
@@ -118,6 +118,14 @@ variable "database_flags" {
   default = []
 }
 
+variable "active_directory_config" {
+  description = "Active domain that the SQL instance will join."
+  type = list(object({
+    domain = string
+  }))
+  default = []
+}
```
Comment (bharathkkb) [inline modules/mssql/variables.tf:127]: Can we just make this a string `active_directory_config_domain` since that is the only supported value?

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @vponnam

File: modules/mssql/variables.tf (hunk 1 above)
Comment (bharathkkb) [reply modules/mssql/variables.tf:127]: Gotcha, in that case lets use `map(string)` and use lookup for relevant values so its extensible? My reasoning for the ask is TF does not support defaults on objects and I want to minimize breaking changes to the module in the future.

Comment (vponnam) [reply modules/mssql/variables.tf]: Sounds good, submitted the proposed change.

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀
Unfortunately it looks like some of our CI checks failed. See the [Contributing Guide](../blob/master/CONTRIBUTING.md) for details.
- ⚠️check_terraform
 Failed Terraform check. More details below. 
 ```bash 
Running terraform fmt
modules/mssql/variables.tf
--- old/modules/mssql/variables.tf
+++ new/modules/mssql/variables.tf
@@ -120,8 +120,8 @@
 
 variable "active_directory_config" {
   description = "Active domain that the SQL instance will join."
-  type = map(string)
-  default = {}
+  type        = map(string)
+  default     = {}
 }
 
 variable "user_labels" {
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
modules/mssql/main.tf
--- old/modules/mssql/main.tf
+++ new/modules/mssql/main.tf
@@ -105,7 +105,7 @@
     dynamic "active_directory_config" {
       for_each = var.active_directory_config
       content {
-        domain =  lookup(var.active_directory_config, "domain", null)
+        domain = lookup(var.active_directory_config, "domain", null)
       }
     }
 
Error: terraform fmt failed with exit code 3
Check the output for diffs and correct using terraform fmt <dir>
 ```

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — DISMISSED** [review]
LGTM, can you also bump the min provider version here?

https://github.com/terraform-google-modules/terraform-google-sql-db/blob/c51bf296e392fca246aae1c9ba4299a5a97ef274/modules/mssql/versions.tf#L23

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vponnam
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Review by bharathkkb — APPROVED** [review]

**Pull request merged**